### PR TITLE
Implement trip share endpoints

### PR DESCRIPTION
**Pull request by jdudycz**

### CODE REVIEW

File: bcycle-backend/Controllers/ShareController.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+        public ShareController(ShareService shareService, TripService tripService)
+        {
+            _shareService = shareService;
+            _tripService = tripService;
+        }
+
+        // GET /api/share/group-trip/:sharingId
+        [HttpGet("group-trip/{sharingId:guid}")]
+        public async Task<ActionResult<ResultContainer<GroupTripShareView>>> GetGroupTrip(Guid sharingId)
```
Comment (K-Wojciechowski) [inline bcycle-backend/Controllers/ShareController.cs:28]: 1. There are already endpoints for this in `GroupTripController`/`TripController`, please delete the old endpoints
2. `[AllowAnonymous]` is missing to make this endpoint accessible without login

File: bcycle-backend/Controllers/ShareController.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        // GET /api/share/trip/:tripId
+        [HttpGet("trip/{sharingId:guid}")]
+        public async Task<ActionResult<ResultContainer<PrivateTripShareView>>> GetPrivateTrip(Guid sharingId)
+        {
+            var privateTripInfo = await _shareService.GetPrivateTripAsync(sharingId);
+            if (privateTripInfo == null) return NotFound();
+            return new ResultContainer<PrivateTripShareView>(privateTripInfo);
+        }
+
+        // POST /api/share/:tripId
```
Comment (K-Wojciechowski) [inline bcycle-backend/Controllers/ShareController.cs:44]: Why not leave this in `TripsController`?

File: bcycle-backend/Models/Entities/GroupTrip.cs [hunk 3]
```diff
@@ -52,15 +49,12 @@ public void Update(GroupTripRequest data)
                 TripCode = TripCode,
                 StartDate = StartDate,
                 EndDate = EndDate,
-                SharingUrl = GetSharingUrl(urlBase, groupTripSharePrefix),
```
Comment (K-Wojciechowski) [inline bcycle-backend/Models/Entities/GroupTrip.cs:55]: We can remove `SharingGuid` from the DB entity as well.

**Review by K-Wojciechowski — CHANGES_REQUESTED** [review]

File: bcycle-backend/Controllers/ShareController.cs (hunk 1 above)
Comment (jdudycz) [reply bcycle-backend/Controllers/ShareController.cs:28]: The data structure  that this endpoint return is modified to correspond to the schema that web is expecting (thats from where the `ShareView` suffix in the return class name comes from). `GroupTripController`/`TripController` return the trip data in a more raw/unmodified format, to be consumed by the mobile app. `[AllowAnonymous]`  would be required if the controller was annotated by `[Authorize]`. Both GET endpoints are meant to be public, but now i see that i forgot to annotate POST and DELETE endpoints to require authorization  after i moved them here. Gonna fix that.

Comment (K-Wojciechowski) [reply bcycle-backend/Controllers/ShareController.cs:28]: Nothing will use these two endpoints anymore:

```cs
        // GET /api/trips/:guid
        [HttpGet("{guid:guid}")]
        [AllowAnonymous]
        public async Task<ActionResult<ResultContainer<TripResponse>>> GetPublic(Guid guid) {}

        // GET /api/group-trips/:guid
        [AllowAnonymous]
        [HttpGet("{guid:guid}")]
        public async Task<ActionResult<ResultContainer<GroupTripResponse>>> Get(Guid guid) {}
```

those endpoints can be removed.

File: bcycle-backend/Controllers/ShareController.cs (hunk 2 above)
Comment (jdudycz) [reply bcycle-backend/Controllers/ShareController.cs:44]: The truth is, i wasn't completely sure about this change, like 55:45. It is sharing related as well as trip related, so responsibility wise it fits both. On the other hand trip controller had a lot more methods that share controller did at this point. Also putting that in share controller would make the url prettier. Taking all of it into account i decided to go for the `ShareController`

File: bcycle-backend/Models/Entities/GroupTrip.cs (hunk 3 above)
Comment (jdudycz) [reply bcycle-backend/Models/Entities/GroupTrip.cs:55]: Done

File: bcycle-backend/Controllers/ShareController.cs (hunk 1 above)
Comment (jdudycz) [reply bcycle-backend/Controllers/ShareController.cs:28]: Oh, that's what you meant. I missed them. Done.

**Review by K-Wojciechowski — APPROVED** [review]

**Pull request merged**